extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableParquetLocationKey.java

```diff
@@ -48,4 +61,45 @@ public String getImplementationName() {
     public ParquetInstructions readInstructions() {
         return readInstructions;
     }
+
+    /**
+     * Precedence-wise this implementation compares {@code order}, then {@code dataSequenceNumber}, then
+     * {@code fileSequenceNumber}, then {@code pos}, then applies a {@link PartitionsComparator} to {@code partitions}
+     * and finally {@code uri}.
+     *
+     * @inheritDoc
+     */
+    @Override
+    public int compareTo(@NotNull final TableLocationKey other) {
+        if (other instanceof IcebergTableParquetLocationKey) {
+            final IcebergTableParquetLocationKey otherTyped = (IcebergTableParquetLocationKey) other;
+            int comparisonResult;
+            if ((comparisonResult = Integer.compare(order, otherTyped.order)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = compareNullable(dataSequenceNumber, otherTyped.dataSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = compareNullable(fileSequenceNumber, otherTyped.fileSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = compareNullable(pos, otherTyped.pos)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = PartitionsComparator.INSTANCE.compare(partitions, otherTyped.partitions)) != 0) {
+                return comparisonResult;
+            }
+            return uri.compareTo(otherTyped.uri);
+        }
+        throw new ClassCastException("Cannot compare " + getClass() + " to " + other.getClass());
+    }
+
+    private static int compareNullable(final Long first, final Long second) {
+        if (first == null || second == null) {
+            // Cannot compare, treat them as equal, so we can continue to the next comparison
+            return 0;
```

Comment:
This, at a minimum, this breaks the transitivity assumption of comparable. Say, `a = null, b = 1, c = 2`, you might say:


* a == b is true
* a == c is true
* therefore b == c is true (oops!)

I think we'll need to pick a direction; either nulls come first, or nulls come last.

In that vein, we can probably use primitives and set them to `Long.MIN_VALUE` or `Long.MAX_VALUE` in the constructor and use a normal primitive comparison.

Reply:
I would prefer keeping them at the end then